src/core/chain/requesthelper.ts

```diff
@@ -34,19 +34,19 @@ export class RequestHelper {
      * @returns {Promise<InvokeResult>}
      * @memberof RequestHelper
      */
-    public invokeRequest(chainMethod: ChainMethod, params: Object, userId: string, invokeAlways = false, transientMap?: TransientMap): Promise<InvokeResult | any> {
+    public invokeRequest(chainMethod: ChainMethod, params: Object, userId?: string, invokeAlways = false, transientMap?: TransientMap): Promise<InvokeResult | any> {
         const args = [JSON.stringify(params)];
 
         return this.hlfClient
             .invoke(chainMethod, args, transientMap)
             .then((response) => {
                 Log.hlf.debug('Invoke successfully executed: ', response);
-                this.eventService.triggerSuccess(userId, chainMethod, params);
+                this.eventService.triggerSuccess(userId || 'UNKNOWN_USER', chainMethod, params);
```

Comment:
I would just not trigger the success on the eventService when the user is not known